server/database/migrations/20211222089010_create_table_user_personalization.ts

```diff
@@ -0,0 +1,23 @@
+import Knex from 'knex';
+
+import { User, UserPersonalization } from '../../src/models';
+import { createTable, dropTable } from '../utils';
+
+exports.up = async (knex: Knex) => {
+  await createTable(knex, UserPersonalization.tableName, (tb) => {
+    tb.string('id').notNullable().primary();
+
+    tb.string('user_id')
+      .notNullable()
+      .references('id')
+      .inTable(User.tableName)
+      .index(`index_${UserPersonalization.tableName}_user_id`)
+      .onDelete('CASCADE');
+    tb.text('type').nullable();
```

Comment:
Should `type` be `notNullable`?

When would it be set to NULL?

Reply:
My original understanding was that all non-primary key and non-foreign key constraints should be nullable.
For example, for a user, email is optional and hence [nullable](https://github.com/OMSCentral/omscentral/blob/69338c9ccf1fb6fae5b4ba5df57ce8de6d190822/server/database/migrations/20191116001050_create_table_user.ts#L14).

However, for user personalization, which is personalization data for a user, the type and value will not be nullable because other than type and value there is nothing else to store in terms of personalization for a given user.
Good point. 
As a next step, I will make both type and value to be nullable.
Hope the understanding that the same argument applies to value makes sense.
Thanks!

Reply:
Done1